packages/components/src/spectrum/CheckboxGroup.tsx

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable react/no-array-index-key */
+import { isElementOfType } from '@deephaven/react-hooks';
+import React, { ReactNode, useMemo } from 'react';
+import {
+  Checkbox,
+  CheckboxGroup as SpectrumCheckboxGroup,
+  SpectrumCheckboxGroupProps,
+} from '@adobe/react-spectrum';
+import { ensureArray } from '@deephaven/utils';
+
+export type CheckboxGroupProps = {
+  children: ReactNode;
+} & Omit<SpectrumCheckboxGroupProps, 'children'>;
+
+/**
+ * Augmented version of the Spectrum CheckboxGroup component that supports
+ * primitive item children.
+ */
+export function CheckboxGroup({
+  children,
+  ...props
+}: CheckboxGroupProps): JSX.Element {
+  const wrappedChildren = useMemo(
+    () =>
+      ensureArray(children).map((child, index) =>
+        isElementOfType(child, Checkbox) ? (
+          React.cloneElement(child, {
+            key: `${index}-${String(child)}`,
+            value: `${index}-${String(child)}`,
+          })
+        ) : (
+          <Checkbox
+            key={`${index}-${String(child)}`}
+            value={`${index}-${String(child)}`}
```

Comment:
Why add `index` to value? This is going to impact selected values passed to `onChange`

Reply:
Also, I don't think we need `index` for the key since checkbox values would be expected to be unique already. eslint actually complained for me when I tried adding an index in my testing.